src/modules/qt/filter_gpstext.yml

```diff
@@ -143,6 +143,19 @@ parameters:
     mutable: yes
     widget: text
     
+  - identifier: swap_180_meridian
```

Comment:
Why would this be an option? Shouldn't this be the only way it works?

Reply:
In theory yes, in practice because I didn't implement this from the begining I am basically working around the issue (instead of finding a 100% working fix - which I don't know how to/and also would probably need a lot more changes; I've seen bigger projects having major issues with the 180 meridian so I think this is great compared to before). I basically shifting all longitudes around starting at file read, process everything with the regular routines (that are not aware if values were shifted or not) and then when displaying data to the output I shift everything back on a need bassis.

I want to leave this to the user because it's not something a lot of people would use/need, and it only works correctly if the track doesn't both cross 180 and 0 meridians. So if someone does a full circle around the globe it will still fail.

This was mostly implemented for the graphic part bcause it just looked unnaceptable even if 1 single point crossed the 180:
![image](https://user-images.githubusercontent.com/3215812/181015057-a5ca0896-40ac-4e80-b481-3a1b406061cd.png)

Reply:
I would like to see if we can find a general solution for this and exclude the user option. I am concerned that it will create confusion and questions from users.

I think it is a common convention to check the line length and always draw the shortest line possible between two points - assuming that there is never more than 180 degrees between two points.

Does your code also handle map edges? In the case of a whole world map, a line might have to be broken into two segments to wrap off the map and continue on the other side (rather than drawing a line across the whole map.

Reply:
> I would like to see if we can find a general solution for this and exclude the user option. I am concerned that it will create confusion and questions from users.

Hmm, I could check if there's points both between 179 and 180 and 180 and -179 and enable it automatically. I'm worried about a full map circle though... I guess I could exclude the swap if there are points near 0.

>I think it is a common convention to check the line length and always draw the shortest line possible between two points - assuming that there is never more than 180 degrees between two points.

The way I draw is I first compute the gps bounds so a whole world would simply draw between -180,-90 and 180,90 then for each point (in draw_main_line_graph()) I compute it's location as a percentage between those edges and then compute that percentage of the drawing rect (in the get_gpspoint_to_rect() call)  and do a p.drawLine() between 2 consecutive points. This is actually where the lineF.intersects() is needed, for efficiency, I only draw points if the segment intersects the drawing rect.

>Does your code also handle map edges? 

It should handle edges but just as a result because of how I draw the lines.

>In the case of a whole world map, a line might have to be broken into two segments to wrap off the map and continue on the other side (rather than drawing a line across the whole map.

This is not treated and is the main reason I did the swap_180 workaround. 
I don't really have a good solution for this, what you're suggesting makes sense theoretically, but how would I check for that reliably? Keep in mind there are ~3600 pairs of points per hour of standard 1s GPS tracking and I need to check this every frame (because of the follow dot feature) so even a simple formula would affect export time (which again, I'd like to keep low, I'm currently getting realtime exporting speed on x265 and all core usage with close to100% cpu which I'm very happy with - and even ~3x on hardware encoding).
And to be honest, this is rather a niche situation, the most common usage would be a real gps track of <2hours at 1s interval over a short location which very likely stays on land. I don't think we should focus in detail on something as extreme as random points on the entire globe, at least not without fixing other more visible issues first.

Reply:
I've been thinking about this and I think removing the option might actually work for most people so I'll try to automatically detect if a swap is needed and just do it at file read regardless of anything else.

Reply:
Automated this and removed user option in commit c954cbe
I have a single test .gpx file found on openstreetmap but seems to pass ok. Regular files are also left as is.